ui/app/pages/settings/advanced-tab/advanced-tab.component.js

```diff
@@ -351,6 +355,74 @@ export default class AdvancedTab extends PureComponent {
     )
   }
 
+  handleIpfsGatewayChange (url) {
+    const { t } = this.context
+
+    this.setState(() => {
+      let ipfsGatewayError = ''
+
+      try {
+        const urlObj = new URL(url) // check if is valid url
+        if (!urlObj.host) {
+          throw new Error('invalid url : no host')
```

Comment:
Instead of throwing here, can we set the `ipfsGatewayError` state?

Reply:
It will get caught by the `try` and do so in the `catch` block.